### DISCUSSION TITLE
Configuring SMTP for Office 365

### DISCUSSION THREAD

**Discussion by routerino** [outline/outline#1901] [Self Hosting]
Hi, has anyone had success configuring SMTP for office 365?

Every time I try I get the following upon sending an email:
`(node:27) UnhandledPromiseRejectionWarning: Error: 139764591164744:error:1408F10B:SSL routines:ssl3_get_record:wrong version number:../deps/openssl/openssl/ssl/record/ssl3_record.c:332:`

It seems like outline is trying to negotiate over SSL instead of STARTTLS, and I can't find any settings to force STARTTLS.

**Comment by tommoor** [comment]
I'm afraid I don't have an outlook server to test against, If you're able to set up some test account details we could potentially try and debug this

**Reply by routerino** [reply]
I'll see what I can do, but I think the red tape to do so for my company would make it troublesome.

I suspect the difference is gmail requires both SSL and STARTTLS, whereas office 365 specifically requires disabling SSL if you're using STARTTLS (and has no SSL option):

![image](https://user-images.githubusercontent.com/45954722/108490619-09b0c380-72f7-11eb-980d-4c913dce0996.png)

I'll do some digging, but I believe I found something in mailer.js that forces SSL to be on if the production flag is set?

**Reply by routerino** [reply]
Gmail settings for comparison:

![image](https://user-images.githubusercontent.com/45954722/108490893-5ac0b780-72f7-11eb-98d8-d6948e53d288.png)

**Reply by routerino** [reply]
Post #3 :).

Here's someone posting what works for them with nodemailer (which is what outline uses I think?)

    var transporter = nodemailer.createTransport({
        host: 'smtp.office365.com', // Office 365 server
        port: 587,     // secure SMTP
        secure: false, // false for TLS - as a boolean not string - but the default is false so just remove this completely
        auth: {
            user: username,
            pass: password
        },
        tls: {
            ciphers: 'SSLv3'
        }
    });

I think it's the TLS cipher as *well* as the inability to set the secure flag to false (in production) that's breaking it.

**Reply by breakingflower** [reply]
@routerino , did you manage to get a solution to this?

**Comment by sarg3nt** [comment]
I have a feeling I'm having basically the same problem but with Apple's mail service.
@tommoor I can post the configs I've tried and get you ap app password for my apple mail account if that would help trouble shooting this.

**Reply by sarg3nt** [reply]
@routerino , I got it working with Apple Mail, not sure if it will help you or not.
I set `SMTP_SECURE=false` and `SMTP_TLS_CIPHERS=TLSv1.2` and that was the trick for me.
Note, it will not send edits a user makes to themselves (makes sense) so to test I had to edit a document as another user.  I'm sure you knew that but FYI in case you didn't.

Here's my full config for Apple mail (password redacted of course)

**Note:** I'm using a custom email domain with Apple mail so my username is the username of my main icloud email account but the smtp from and reply are to the custom domain

```
SMTP_HOST=smtp.mail.me.com
SMTP_PORT=587
SMTP_USERNAME=dsarg3
SMTP_PASSWORD=<redacted>
SMTP_FROM_EMAIL=admin@sarg3.net
SMTP_REPLY_EMAIL=admin@sarg3.net
SMTP_TLS_CIPHERS=TLSv1.2
SMTP_SECURE=false
```

**Reply by tommoor** [reply]
I just created a new section in the SMTP docs for known working configurations and added your example:

https://wiki.generaloutline.com/s/hosting/doc/smtp-cqCJyZGMIB#h-examples
